src/material/core/option/option.scss

```diff
@@ -131,3 +132,28 @@
 .mat-mdc-option-active::before {
   content: '';
 }
+
+// Re-apply the `border` and `border-radius` for strong focus. Override the MDC styles which come
+// list. MDC uses the `::before` pseudo element for the default focus/hover/selected state. The MDC
+// styles normally override the strong focus indicator styles. Increase specificity to override MDC.
+// Fix issue where strong focus indicator is not present for selected option (#26801).
+.mat-mdc-option-active.mat-mdc-focus-indicator::before {
```

Comment:
I'm a little bit concerned that we duplicate a bunch of styles here. Could we follow the same approach as the `mat-list-option` which uses the same styles and a dedicated DOM node for the focus indicator? See https://github.com/angular/components/blob/main/src/material/list/list-option.html#L84

Reply:
Okay that's the good point. I can try adding an element for the focus indicator. Adding another DOM node could cause a performance regression, but I don't the difference would be noticeable unless there are thousands of options.

Reply:
I think that there are more pressing performance issues with `mat-option`...